Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/analytics/SettingsPrivacyPreservingAnalyticsState.kt

```diff
@@ -0,0 +1,35 @@
+package de.rki.coronawarnapp.ui.settings.analytics
+
+import android.content.Context
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.datadonation.analytics.common.Districts
+import de.rki.coronawarnapp.datadonation.analytics.common.labelStringRes
+import de.rki.coronawarnapp.server.protocols.internal.ppdd.PpaData
+
+data class SettingsPrivacyPreservingAnalyticsState(
+    val isAnalyticsEnabled: Boolean,
+    val ageGroup: PpaData.PPAAgeGroup,
+    val federalState: PpaData.PPAFederalState,
+    val district: Districts.District?
+) {
+    fun getAgeGroupRowBodyText(context: Context) =
+        context.getString(ageGroup.labelStringRes)
+
+    fun getDistrictRowVisibility() =
+        federalState != PpaData.PPAFederalState.FEDERAL_STATE_UNSPECIFIED
+
+    fun getDistrictRowBodyText(context: Context) =
+        district?.districtName ?: context.getString(R.string.analytics_userinput_district_unspecified)
+
+    fun getFederalStateRowBodyText(context: Context) =
+        context.getString(federalState.labelStringRes)
+
+    fun getSettingsPpaSwitchRowState() =
```

Comment:
I think it is better here to call it `isSettingsPpaSwitchOn`. it is not actually getting a state. it is just boolean

Reply:
👍